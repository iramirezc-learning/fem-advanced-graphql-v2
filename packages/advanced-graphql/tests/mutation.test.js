const createTestServer = require('./helper')
const { UPDATE_SETTINGS } = require('./queries')

const mockUser = {
  id: 1,
  email: 'user@example.com',
  avatar: 'http://placeholder.com',
  verified: true,
  createdAt: new Date(2021, 2, 11).getTime(),
  role: 'MEMBER'
}

describe('mutations', () => {
  test('updateSettings', async () => {
    const { mutate } = createTestServer({
      user: mockUser,
      models: {
        Settings: {
          updateOne: jest.fn((_, input) => ({
            id: 1,
            ...input
          }))
        },
        User: {
          findOne: jest.fn(() => mockUser)
        }
      }
    })

    const res = await mutate({
      mutation: UPDATE_SETTINGS,
      variables: {
        input: {
          theme: 'DARK',
          emailNotifications: true,
          pushNotifications: true
        }
      }
    })
    expect(res).toMatchSnapshot()
  })
})
