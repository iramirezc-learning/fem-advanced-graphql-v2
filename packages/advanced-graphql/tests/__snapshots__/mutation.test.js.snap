// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations updateSettings 1`] = `
Object {
  "data": Object {
    "updateSettings": Object {
      "emailNotifications": true,
      "id": "1",
      "pushNotifications": true,
      "theme": "DARK",
      "user": Object {
        "email": "user@example.com",
        "id": "1",
      },
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
